bin/compile

```diff
@@ -1,2 +1,159 @@
+			Please try again and to see if the error resolves itself.
+		EOF
+		exit 1
+	fi
+
+	local manifest_digest
+	manifest_digest=$(jq -r '.manifests[0].digest' "${temp_dir}/index.json")
+	local manifest_file="${temp_dir}/blobs/sha256/${manifest_digest#sha256:}"
+
+	# Prepare the target directory and extract layers from the manifest file
+	mkdir -p "${CNB_BUILDPACK_DIR}"
+	jq -r '.layers[].digest' "${manifest_file}" | while read -r layer_digest; do
+		# Extract each layer, removing the nested prefix directories (e.g. `cnb/buildpacks/heroku_dotnet/0.1.6`)
+		tar --strip-components=4 -xf "${temp_dir}/blobs/sha256/${layer_digest#sha256:}" -C "${CNB_BUILDPACK_DIR}" 2>/dev/null
+	done
+}
+
+# Function to format CNB output, rewriting section prefixes to classic
+# buildpack output style.
+# TODO: Remove when the CNB supports printing classic output style natively.
+format_cnb_output() {
+	while IFS= read -r line; do
+		local color_code='' # Holds any detected initial color code
+		local clean_line="${line}"
+
+		# Detect and store an ANSI color code at the start of the line, if present
+		if [[ "${line}" =~ ^($'\x1b['[0-9;]*'m') ]]; then
+			color_code="${BASH_REMATCH[1]}"
+			clean_line="${line#"${color_code}"}" # Remove color code for processing
+		fi
+
+		# Set prefix and content based on line pattern
+		case "${clean_line}" in
+			"##"*)
+				prefix='----->'
+				content="${clean_line:2}"
+				;;
+			"-"*)
+				prefix='----->'
+				content="${clean_line:1}"
+				;;
+			"  -"*)
+				prefix='      '
+				content="${clean_line:3}"
+				;;
+			"!"*)
+				prefix=' !    '
+				content="${clean_line:1}"
+				;;
+			"")
+				prefix=''
+				content="${clean_line}"
+				;;
+			*)
+				prefix='   '
+				content="${clean_line}"
+				;;
+		esac
+
+		# Output formatted line with original color code, custom prefix, and content
+		echo "${color_code}${prefix}${content}"
+	done
+}
+
+# Run build with formatted output
+run_build() {
+	"${CNB_BUILDPACK_DIR}/bin/build" "${CNB_BUILD_DIR}/layers" "${CNB_BUILD_DIR}/platform" "${CNB_BUILD_DIR}/buildpack_plan.toml" | format_cnb_output
+}
+
+process_env_files() {
+	rm -f "${PROFILE_PATH}"
+	find "${CNB_BUILD_DIR}/layers" -type f \( -path "*/env/*" -o -path "*/env.launch/*" \) | while read -r file; do
+		process_env_file "${file}" >>"${PROFILE_PATH}"
+	done
+}
+
+# Remove build layers from build directory to reduce slug size
+remove_build_layers() {
+	for item in "sdk" "nuget-cache"; do
+		rm -rf "${CNB_BUILD_DIR}/layers/${item}" "${CNB_BUILD_DIR}/layers/${item}.toml"
+	done
+}
+
+process_env_file() {
+	env_file="$1"
+	var_name=$(basename "${env_file}" | cut -d '.' -f 1)
+	operation=$(basename "${env_file}" | cut -d '.' -f 2)
+	var_value=$(cat "${env_file}")
+	case "${operation}" in
+		"override") echo "export ${var_name}=\"${var_value}\"" ;;
+		"prepend") echo "export ${var_name}=\"${var_value}:\$${var_name}\"" ;;
+		"append") echo "export ${var_name}=\"\$${var_name}:${var_value}\"" ;;
```

Comment:
prepend+append will currently leave a leading/trailing colon if the value wasn't set already. For some env vars this empty element in the colon-delimited list will actually has a special meaning (eg for PATH it means current directory).

I'd recommend using the `${var:+VALUE_IF_SET}` parameter expansion feature, similar to:
https://github.com/heroku/heroku-buildpack-python/pull/1645

Reply:
Addressed in https://github.com/heroku/heroku-buildpack-dotnet/pull/14/commits/32cbca86c5d8423d2cb9c58cef396054742ad88f